controllers/providers/pxc/applier.go

```diff
@@ -116,10 +116,14 @@ func (p *applier) Engine() error {
 	p.PerconaXtraDBCluster.Spec.PXC.PodSpec.Affinity = &pxcv1.PodAffinity{
 		Advanced: common.DefaultAffinitySettings().DeepCopy(),
 	}
+	hasDBSpecChanged := func() bool {
+		return p.DB.Status.ObservedGeneration > 0 && p.DB.Status.ObservedGeneration != p.DB.Generation
+	}
 	// We preserve the settings for existing DBs, otherwise restarts are seen when upgrading Everest.
+	// Additionally, we also need to check for the spec changes, otherwise the user can never voluntarily change the resource setting.
 	// TODO: Remove this once we figure out how to apply such spec changes without automatic restarts.
 	// See: https://perconadev.atlassian.net/browse/EVEREST-1413
-	if p.DB.Status.Status == everestv1alpha1.AppStateReady {
+	if p.DB.Status.Status == everestv1alpha1.AppStateReady && !hasDBSpecChanged() {
```

Comment:
It should work fine since operator always updates status as a last step of reconciliation, so LGTM.  

should we apply the check for all engines affinity and also for the pxc resources? Basically, it means everywhere we have this
```
// TODO: Remove this once we figure out how to apply such spec changes without automatic restarts.
```

Reply:
> should we apply the check for all engines affinity
Affinity settings cannot be updated by the user yet, so I think it's fine to skip it. The reason we applied it for PXC resource settings is because it can be changed by the user, so we need to be able to detect spec changes.. So, for affinity, I don't think it matters. Moreover, I'd eventually like to get rid of all these manual if-else checks and implement a better mechanism, which takes the current and previous Everest version into account..

> also for the pxc resources?
Is there any other pxc resources you're referring to other than the one fixed in this PR?

Reply:
@mayankshah1607 I mean, I was confused bc the check is applied to the affinity setting now, look at the next line:
```
 pxc.Spec.PXC.PodSpec.Affinity = p.currentPerconaXtraDBClusterSpec.PXC.Affinity
```
it should've been applied to the line 102 instead. Then I would've no more questions :)

Reply:
Ohh crap. I messed up something when refactoring, thanks for pointing out..
Yeah, it should have been for the resources only 🙈